x-pack/plugins/elastic_assistant/server/routes/knowledge_base/get_knowledge_base_indices.test.ts

```diff
@@ -38,7 +72,83 @@ const mockFieldCaps = {
         metadata_field: false,
         searchable: true,
         aggregatable: false,
-        indices: ['semantic-index-1', 'semantic-index-2'],
+        indices: ['gtr-with-semantic-1'],
+      },
+    },
+    ai_embeddings: {
+      unmapped: {
+        type: 'unmapped',
+        metadata_field: false,
+        searchable: false,
+        aggregatable: false,
+        indices: [
+          '.ds-.items-default-2024.11.12-000001',
+          '.ds-.lists-default-2024.11.12-000001',
+          '.ds-logs-endpoint.alerts-default-2024.11.12-000001',
+          '.ds-logs-endpoint.events.process-default-2024.11.12-000001',
+          '.ds-metrics-endpoint.metadata-default-2024.11.12-000001',
+          '.ds-metrics-endpoint.policy-default-2024.11.12-000001',
+          '.internal.alerts-default.alerts-default-000001',
+          '.internal.alerts-ml.anomaly-detection-health.alerts-default-000001',
+          '.internal.alerts-ml.anomaly-detection.alerts-default-000001',
+          '.internal.alerts-observability.apm.alerts-default-000001',
+          '.internal.alerts-observability.logs.alerts-default-000001',
+          '.internal.alerts-observability.metrics.alerts-default-000001',
+          '.internal.alerts-observability.slo.alerts-default-000001',
+          '.internal.alerts-observability.threshold.alerts-default-000001',
+          '.internal.alerts-observability.uptime.alerts-default-000001',
+          '.internal.alerts-security.alerts-default-000001',
+          '.internal.alerts-stack.alerts-default-000001',
+          '.internal.alerts-transform.health.alerts-default-000001',
+          'gtr-1',
+          'gtr-with-semantic-1',
+          'metrics-endpoint.metadata_current_default',
+        ],
+      },
+      semantic_text: {
+        type: 'semantic_text',
+        metadata_field: false,
+        searchable: true,
+        aggregatable: false,
+        indices: ['gtr-with-bug', 'search-elastic-security-docs'],
+      },
+    },
+    semantic_text: {
+      unmapped: {
```

Comment:
I think you have too much mock data here that is superfluous. By changing `mockData` to the following, the tests still prove the fix:

```
const mockFieldCaps = {
  indices: [
    'gtr-1',
    'gtr-with-bug',
    'gtr-with-semantic-1',
    'metrics-endpoint.metadata_current_default',
    'search-elastic-security-docs',
  ],
  fields: {
    content: {
      semantic_text: {
        indices: ['gtr-with-semantic-1'],
      },
    },
    ai_embeddings: {
      semantic_text: {
        indices: ['gtr-with-bug', 'search-elastic-security-docs'],
      },
    },
    semantic_text: {
      semantic_text: {
        indices: ['search-elastic-security-docs'],
      },
    },
  },
};
```

Reply:
Ah, webstorm is so slow. I see this messes up the type. Feel free to disregard. But maybe we can cut down on those indices, especially the `unmapped` which seem unused

Reply:
Yeah, I just copied mappings from my local kibana to make it as close as possible to real setup :-) Will simplify it though.